### PR TITLE
chore: update abci++ spec

### PR DESCRIPTION
**Pull request by shotonoff**
## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: spec/abci++/abci++_methods_002_draft.md [hunk 1]
```diff
@@ -61,22 +61,23 @@ title: Methods
...
+    | app_state_bytes     | bytes                                                                                                                                  | Serialized initial application state. JSON bytes.             | 5            |
+    | initial_height      | int64                                                                                                                                  | Height of the initial block (typically `1`).                  | 6            |
+    | initial_core_height | int64                                                                                                                                  | Height of the initial core height.                            | 7            |
 
 * **Response**:
 
     | Name             | Type                                         | Description                                     | Field Number |
-    |------------------|----------------------------------------------|-------------------------------------------------|--------------|
+------------------|------------------|----------------------------------------------|-------------------------------------------------|--------------|
```
Comment (thephez) [inline spec/abci++/abci++_methods_002_draft.md:77]: I think this should actually be (it doesn't render the table properly currently):
```suggestion
|------------------|----------------------------------------------|-------------------------------------------------|--------------|
```

File: spec/abci++/abci++_methods_002_draft.md [hunk 2]
```diff
@@ -401,12 +402,12 @@ Note that, if _p_ has a non-`nil` _validValue_, Tendermint will use it as propos
...
     | height               | int64                                       | The height of the proposed block.                                                                              | 5            |
     | time                 | [google.protobuf.Timestamp](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp) | Timestamp included in the proposed block.  | 6            |
     | next_validators_hash | bytes                                       | Merkle root of the next validator set.                                                                         | 7            |
-    | proposer_address     | bytes                                       | [Address](../core/data_structures.md#address) of the validator that created the proposal.                      | 8            |
+    | proposer_pro_hx_hash | bytes                                       | [ProTxHash](../core/data_structures.md#protxhash) of the validator that created the proposal.                  | 8            |
```
Comment (thephez) [inline spec/abci++/abci++_methods_002_draft.md:410]: Typo?
```suggestion
    | proposer_pro_tx_hash | bytes                                       | [ProTxHash](../core/data_structures.md#protxhash) of the validator that created the proposal.                  | 8            |
```

File: spec/abci++/abci++_methods_002_draft.md [hunk 3]
```diff
@@ -468,25 +469,25 @@ When a validator _p_ enters Tendermint consensus round _r_, height _h_, in which
...
+    | hash   | bytes | The header hash of the proposed block that the vote extensions is to refer to. | 1            |
+    | height | int64 | Height of the proposed block (for sanity check).                               | 2            |
 
 * **Response**:
 
-    | Name              | Type  | Description                                   | Field Number |
-    |-------------------|-------|-----------------------------------------------|--------------|
-    | vote_extension    | bytes | Optional information signed by by Tendermint. | 1            |
+    | Name            | Type  | Description                                   | Field Number |
+-----------------|-------------------|-------|-----------------------------------------------|--------------|
```
Comment (thephez) [inline spec/abci++/abci++_methods_002_draft.md:480]: I think this should actually be (it doesn't render the table properly currently):
```suggestion
|-------------------|-------|-----------------------------------------------|--------------|
```

**Review by thephez — COMMENTED** [review]
Just a few suggestions

**Review by shotonoff — COMMENTED** [review]
@thephez thank you for your suggestions, all accepted

File: spec/abci++/abci++_methods_002_draft.md [hunk 4]
```diff
@@ -61,22 +61,23 @@ title: Methods
...
+    | app_state_bytes     | bytes                                                                                                                                  | Serialized initial application state. JSON bytes.             | 5            |
+    | initial_height      | int64                                                                                                                                  | Height of the initial block (typically `1`).                  | 6            |
+    | initial_core_height | int64                                                                                                                                  | Height of the initial core height.                            | 7            |
 
 * **Response**:
 
     | Name             | Type                                         | Description                                     | Field Number |
-    |------------------|----------------------------------------------|-------------------------------------------------|--------------|
+|------------------|----------------------------------------------|-------------------------------------------------|--------------|
```
Comment (thephez) [inline spec/abci++/abci++_methods_002_draft.md:77]: Still didn't render correctly. This should fix it.
```suggestion
    |------------------|----------------------------------------------|-------------------------------------------------|--------------| 
```

File: spec/abci++/abci++_methods_002_draft.md [hunk 5]
```diff
@@ -468,25 +469,25 @@ When a validator _p_ enters Tendermint consensus round _r_, height _h_, in which
...
+    | hash   | bytes | The header hash of the proposed block that the vote extensions is to refer to. | 1            |
+    | height | int64 | Height of the proposed block (for sanity check).                               | 2            |
 
 * **Response**:
 
-    | Name              | Type  | Description                                   | Field Number |
-    |-------------------|-------|-----------------------------------------------|--------------|
-    | vote_extension    | bytes | Optional information signed by by Tendermint. | 1            |
+    | Name            | Type  | Description                                   | Field Number |
+|-------------------|-------|-----------------------------------------------|--------------|
```
Comment (thephez) [inline spec/abci++/abci++_methods_002_draft.md:480]: Still didn't render correctly. This should fix it.
```suggestion
    |-------------------|-------|-----------------------------------------------|--------------|
```

**Review by thephez — COMMENTED** [review]
There was also an indenting issue preventing those 2 tables from rendering.

**Review by shotonoff — COMMENTED** [review]
@thephez suggested accepted

**Review by thephez — APPROVED** [review]

**Pull request merged**